infra/scoped/backend.tf

```diff
@@ -1,6 +1,4 @@
 terraform {
-  required_version = "= 0.14.2" # Pin to a specific version to avoid accidental upgrading of the statefile
```

Comment:
Unpin the terraform version as we don't have a mechanism so we do not have to explicitly care about which version we are using in the runner script. 

We do not pin terraform in any other project, so this is more familiar to Wellcome developers, although it _will_ cause the TF version to roll forward with all the associated issues.